plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/NullPushdownDataTypeTest.java

```diff
@@ -0,0 +1,169 @@
+            verifyPredicate(queryRunner, session, temporaryRelation, false, false, !connectorExpressionOnly);
+            verifyPredicate(queryRunner, session, temporaryRelation, false, true, true);
+        }
+        return this;
+    }
+
+    private void verifyPredicate(QueryRunner queryRunner, Session session, TemporaryRelation temporaryRelation, boolean isNull, boolean connectorExpression, boolean expectPushdown)
+    {
+        TestCase firstCase = testCases.getFirst();
+        String firstColumnName = "col_0";
+        String withConnectorExpression = connectorExpression ? " OR %s IS NULL".formatted(firstColumnName) : "";
+
+        String queryWithAll = "SELECT " + firstColumnName + " FROM " + temporaryRelation.getName() + " WHERE " +
+                IntStream.range(SPECIAL_COLUMNS, testCases.size())
+                        .mapToObj(column -> getPredicate(column, isNull))
+                        .collect(joining(" AND "))
+                + withConnectorExpression;
+
+        // Closing QueryAssertions would close the QueryRunner
+        QueryAssertions queryAssertions = new QueryAssertions(queryRunner);
+        try {
+            assertPushdown(expectPushdown,
+                    assertResult(isNull ? Optional.of(firstCase.expectedLiteral()) : Optional.empty(),
+                            assertThat(queryAssertions.query(session, queryWithAll))));
+        }
+        catch (AssertionError e) {
+            // if failed - identify exact column which caused the failure
+            for (int column = SPECIAL_COLUMNS; column < testCases.size(); column++) {
+                String queryWithSingleColumnPredicate = "SELECT " + firstColumnName + " FROM " + temporaryRelation.getName() + " WHERE " + getPredicate(column, isNull) + withConnectorExpression;
+                assertPushdown(expectPushdown,
+                        assertResult(isNull ? Optional.of(firstCase.expectedLiteral()) : Optional.empty(),
+                                assertThat(queryAssertions.query(session, queryWithSingleColumnPredicate))));
+            }
+            throw new IllegalStateException("Single column assertion should fail for at least one column, if query of all column failed", e);
+        }
+    }
+
+    private String getPredicate(int column, boolean isNull)
+    {
+        checkArgument(column >= SPECIAL_COLUMNS, "Special columns should not be a part of a predicate, as they are helpers and do not participate in the assertions");
+        String columnName = "col_" + column;
+        checkArgument("NULL".equalsIgnoreCase(testCases.get(column).inputLiteral()));
+        return isNull
+                ? columnName + " IS NULL"
+                : columnName + " IS NOT NULL";
+    }
+
+    private static QueryAssert assertResult(Optional<String> value, QueryAssert assertion)
+    {
+        return value.isPresent()
+                ? assertion.matches("VALUES %s".formatted(value.get()))
+                : assertion.returnsEmptyResult();
+    }
```

Comment:
If we could inline `IS NULL` and `IS NOT NULL` and the assertion could be inlined right ? Ideally for the test data - we for `IS NULL` we get a value and for `IS NOT NULL` we get an empty value - Do we need this dedicated method for each assertion.